### PR TITLE
[Hidden files] - Add support for moving or permanently deleting a hidden folder

### LINKED ISSUE #2402: Default left icons missing from list view in network drives
**Describe the bug**
Network drives shows with missing icons in the list view.

**To Reproduce**
1. Open the application.
2. Click on any *network drive*.
3. Archives left icon are missing from list view.

**Expected behavior**
List of archives with default folder icon to the left to be shown.

**Screenshots**

**Desktop (please complete the following information):**
 - OS Version: [e.g. Windows 10 20H2 19042 6.30]
 - App version: [v0.20.2.0]

**Additional context**
This only happens with *network  drives*, I tried refreshing the location with no luck, though, when selecting another left pane folder / drive from there, the icons are shown for a brief instant before go the other folder / drive. 
On local drives icons are properly shown.

### PR DESCRIPTION
**Pull request by gave92**
For #2303
Fixes #2402
Fixes an issue for which you currently cannot move or permanently delete a hidden folder
Fixes an issue for which no icons are shown in recycle bin *and network paths*

### CODE REVIEW

**Review by yaichenbaum — APPROVED** [review]

File: Files.Launcher/Program.cs [hunk 1]
```diff
@@ -422,11 +422,13 @@ private static async Task ParseFileOperationAsync(AppServiceRequestReceivedEvent
...
 
-                case "MoveToBin":
+                case "DeleteItem":
                     var fileToDeletePath = (string)args.Request.Message["filepath"];
+                    var permanently = (bool)args.Request.Message["permanently"];
                     using (var op = new ShellFileOperations())
                     {
-                        op.Options = ShellFileOperations.OperationFlags.AllowUndo | ShellFileOperations.OperationFlags.NoUI;
+                        op.Options = ShellFileOperations.OperationFlags.NoUI;
+                        if (!permanently) op.Options |= ShellFileOperations.OperationFlags.AllowUndo;
```
Comment (tsvietOK) [inline Files.Launcher/Program.cs:431]: Please, always check your code is match to our [Code style](https://github.com/files-community/Files/wiki/Code-style).

**Review by yaichenbaum — APPROVED** [review]

File: Files/Commands/Paste.cs [hunk 2]
```diff
@@ -202,10 +203,17 @@ await AppInstance.FilesystemViewModel.GetFolderFromPathAsync(destinationPath)
 
                     if (deleted == FilesystemErrorCode.ERROR_UNAUTHORIZED)
                     {
-                        // Try again with DeleteFileFromApp
-                        if (!NativeFileOperationsHelper.DeleteFileFromApp(item.Path))
+                        // Try again with fulltrust process
+                        if (AppInstance.FilesystemViewModel.Connection != null)
```
Comment (duke7553) [inline Files/Commands/Paste.cs:207]: Is there ever a scenario where we want to try again with the *FromApp method?

Comment (gave92) [reply Files/Commands/Paste.cs:207]: Yes, if we want to delete the file permanently we can use deletefilefromapp.
Problem is it doesn't work for folders, the method for folders (removedirectory) only works for empty folders so I'd need to recursively delete the folder contents.
I was too lazy to do that and this is much simpler

**Comment by tsvietOK** [comment]
@gave92 When you insert a hidden folder into another folder, it becomes not hidden.

File: Files/Commands/Paste.cs (hunk 2 above)
Comment (duke7553) [reply Files/Commands/Paste.cs:212]: @gave92 Thanks for the explanation that should suffice for now.

**Review by duke7553 — APPROVED** [review]

**Review by tsvietOK — APPROVED** [review]

**Pull request merged**